dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java

```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
         return new Builder(dbClient);
     }
 
+    /**
+     * Execute the ping statement.
+     *
+     * @return {@code Awaitable} instance to wait for
+     */
+    protected abstract Awaitable<?> execPing();
+
     @Override
     public HealthCheckResponse call() {
-        HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
-                .name(name);
+        HealthCheckResponseBuilder builder = HealthCheckResponse.builder().name(name);
 
         try {
-            dbClient.ping().await(10, TimeUnit.SECONDS);
+            execPing().await(timeoutSeconds, TimeUnit.SECONDS);
             builder.up();
         } catch (Throwable e) {
             builder.down();
             builder.withData("ErrorMessage", e.getMessage());
             builder.withData("ErrorClass", e.getClass().getName());
-            e.printStackTrace();
+            // Do not process logging arguments until it makes sense.
+            if (LOGGER.isLoggable(Level.FINER)) {
```

Comment:
Please do not do this, use 
`Logger.FINER(() -> String.format("...")` or equivalent method with a lambda - it achieves the same thing without additional if statement in the code

Reply:
fixed